crates/uv/src/commands/pip_install.rs

```diff
@@ -116,8 +118,29 @@ pub(crate) async fn pip_install(
     } else if system {
         PythonEnvironment::from_default_python(&cache)?
     } else {
-        PythonEnvironment::from_virtualenv(&cache)?
+        match PythonEnvironment::from_virtualenv(&cache) {
+            Ok(venv) => venv,
+            Err(uv_interpreter::Error::VenvNotFound) => {
+                let path = std::env::current_dir()?.join(".venv");
+                let interpreter = find_default_python(&cache)?;
+
+                writeln!(
+                    printer.stderr(),
+                    "Creating Python {} virtual environment at: {}",
+                    interpreter.python_version(),
+                    path.user_display().cyan()
+                )?;
+
+                if let Some(activation) = commands::venv::activation(&path) {
+                    writeln!(printer.stderr(), "Activate with: {}", activation.green())?;
+                }
+
+                commands::venv::create(&path, interpreter, Prompt::default(), false)?
```

Comment:
As noted by Damian, we should definitely not create this when `--dry-run` is enabled.